wazo_calld/plugins/calls/services.py

```diff
@@ -224,12 +224,26 @@ def originate(self, request):
             channel = self._ari.channels.get(channelId=channel_id)
 
         else:
+            error_message = None
             if 'line_id' in request['source']:
-                endpoint = user.line(request['source']['line_id']).interface()
+                requested_endpoint = user.line(request['source']['line_id'])
+                if not requested_endpoint.is_online(self._ari):
+                    error_message = "Could not dial caller's requested line"
+                endpoint = requested_endpoint.interface()
+
             elif request['source']['all_lines']:
                 endpoint = f"local/{source_user}@usersharedlines"
             else:
-                endpoint = user.main_line().interface()
+                requested_endpoint = user.main_line()
+                if not requested_endpoint.is_online(self._ari):
+                    error_message = "Could not dial caller's main line"
+                endpoint = requested_endpoint.interface()
+
+            if error_message is not None:
+                details = {
+                    'caller_extension': endpoint,
+                }
+                raise CallCreationError(error_message, details=details)
```

Comment:
- I would seek to avoid(as possible) nullable/stateful variables and try to minimize conditionals, such as with `error_message`; if you see an error condition, raising directly leads to more readable code(the exception is generated directly from where the error condition is detected, easier to read and understand from the source code and from a traceback)
- There's duplication of logic between the first(`if`) and third(`else`) conditional branches; the only difference is how the line is identified; the `all_lines` case is the special one, where no specific line is selected and a special interface is used; i think this code can be reorganized for less duplication and clearer logic; special cases that are easy to spot can usually be checked and handled first

Reply:
I'll look into refactoring that

Reply:
Refactored :+1: